lapis/file_provider.py

```diff
@@ -0,0 +1,37 @@
+from lapis.storage import Storage
+
+
+class FileProvider(object):
+
+    __slots__ = ("storages",)
+
+    def __init__(self):
+        self.storages = dict()
+
+    def add_storage_element(self, storage_element: Storage):
+        try:
+            self.storages[storage_element.sitename].append(storage_element)
+        except KeyError:
+            self.storages[storage_element.sitename] = [storage_element]
+
+    def provides_all_files(self, job):
+        """
+        Dummy implementation, to be replaced: if a part of every inputfile of the job is
+        provided by a storage element located on the same site as the drone the job
+        is running on this function returns True
+        :param job:
+        :return:
+        """
+        provided_storages = self.storages.get(job.drone.sitename, None)
```

Comment:
This evaluation is way too late. When a job is already assigned to a drone it means, that it is already going to run there. So why should you at this point in time even care about if all files are provided on that site?

Reply:
There is not supposed to be any kind of evaluation at the moment. In the dummy caching approach that is currently implemented as a place holder, a job is sped up to 80% CPU efficiency if a part of every input file is provided on that side. Therefor the file_provider checks if the job is sped up or not. 
There is no coordination implementation yet as I wanted to figure out the mechanics of the caches before implementing coordination to safe time. But yes, the `file_provider` will be called during scheduling once we start to coordinate.

Reply:
ah great, thanks for explanation :) You could also leave those hints in docstrings to guide us.